src/base-field/base-field.tsx

```diff
@@ -213,7 +213,23 @@ export type BaseFieldProps = WithEnhancedClassName &
          * @default 'below'
          */
         characterCountPosition?: 'below' | 'inline' | 'hidden'
-    }
+    } & (
+        | {
+              supportsStartAndEndSlots?: false
+              endSlot?: never
+              endSlotPosition?: never
+          }
+        | {
+              supportsStartAndEndSlots: true
+              endSlot?: React.ReactElement | string | number
+              /**
+               * This is primarily for `bordered` variants of TextField. When set to `bottom` (the default),
```

Comment:
💭 Looking at the code, it seems like this is "solely" more than "primarily" for `bordered` variants of TextField (we'd need to update this comment to make it more clear).

Reply:
Fair.

Reply:
What I meant by `endSlotPosition` being "solely" for `bordered` variants of TextField is that if I have a `default` version but I still pass `endSlotPosition: fullHeight`, that will make so that what I passed as `endSlot` doesn't get honored:

![Screenshot 2025-03-12 at 15 04 17](https://github.com/user-attachments/assets/b07df643-2a3d-4d11-8972-33e63a6bc214)

![Screenshot 2025-03-12 at 15 04 25](https://github.com/user-attachments/assets/984621d0-c48d-4abc-a992-a3c898d6cd91)

I feel like we should just treat it as `endSlotPosition: bottom`.

Reply:
So update the types again so that if it's not `bordered` then `endSlotPosition` can't be set?